### ISSUE TITLE
Segmentation fault occurs  when loading ONNX v6 model shufflenet_v2

### ISSUE THREAD

**Issue by cxnaive** [opencv/opencv#18898]
##### System information (version)

- OpenCV => 4.5.0
- Operating System / Platform => Ubuntu 20.04
- Compiler => gcc x64

##### Detailed description

when loading onnx net converted from Pytorch. segmentation fault occurs.
model:
https://pytorch.org/hub/pytorch_vision_shufflenet_v2/
onnx_file:

[net.zip](https://github.com/opencv/opencv/files/5582255/net.zip)

and map:
![1](https://user-images.githubusercontent.com/20917336/99951679-729ca400-2db9-11eb-9438-cfc3f311a08e.png)
![2](https://user-images.githubusercontent.com/20917336/99951695-77f9ee80-2db9-11eb-9cfe-2a4099772300.png)
##### Steps to reproduce

```.cpp
cout << "net init begins!" << endl;
net = dnn::readNetFromONNX(modelBin);
cout << "Read done." << endl;
```
##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by asmorkalov** [comment]
The issue reproduced with OpenCV master (4.5.0-211-g0105f8fa38) on Ubuntu 18.04.

**Comment by alalek** [comment]
Segmentation fault or just C++ exception (which is uncatched so std::terminate is calling)?

---

> Segmentation fault

Please provide stacktrace (from debug build).

**Comment by asmorkalov** [comment]
```
net init begins!
[ INFO:0] global /home/alexander/Projects/OpenCV/opencv-master/modules/dnn/src/onnx/onnx_importer.cpp (395) populateNet DNN/ONNX: loading ONNX v6 model produced by 'pytorch':1.5. Number of nodes = 261, inputs = 1, outputs = 1

Program received signal SIGSEGV, Segmentation fault.
__memmove_sse2_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:487
487     ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S: Нет такого файла или каталога.
(gdb) bt
#0  __memmove_sse2_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:487
#1  0x00007ffff7716316 in std::__copy_move<true, true, std::random_access_iterator_tag>::__copy_m<int> (__first=0x555555af8cc0, __last=0x555555af8cbc, __result=0x555555af8cbc) at /usr/include/c++/7/bits/stl_algobase.h:368
#2  0x00007ffff771623c in std::__copy_move_a<true, int*, int*> (__first=0x555555af8cc0, __last=0x555555af8cbc, __result=0x555555af8cbc) at /usr/include/c++/7/bits/stl_algobase.h:386
#3  0x00007ffff7797b7c in std::__copy_move_a2<true, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > > (__first=0, __last=7, __result=7)
    at /usr/include/c++/7/bits/stl_algobase.h:422
#4  0x00007ffff779792a in std::move<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > > (__first=0, __last=7, __result=7) at /usr/include/c++/7/bits/stl_algobase.h:488
#5  0x00007ffff789b99b in std::vector<int, std::allocator<int> >::_M_erase (this=0x7fffffffccc0, __position=7) at /usr/include/c++/7/bits/vector.tcc:157
#6  0x00007ffff7899f27 in std::vector<int, std::allocator<int> >::erase (this=0x7fffffffccc0, __position=7) at /usr/include/c++/7/bits/stl_vector.h:1180
#7  0x00007ffff78837dc in cv::dnn::dnn4_v20201117::ONNXImporter::handleNode (this=0x7fffffffd9f0, node_proto_=...) at /home/alexander/Projects/OpenCV/opencv-master/modules/dnn/src/onnx/onnx_importer.cpp:509
#8  0x00007ffff7882638 in cv::dnn::dnn4_v20201117::ONNXImporter::populateNet (this=0x7fffffffd9f0) at /home/alexander/Projects/OpenCV/opencv-master/modules/dnn/src/onnx/onnx_importer.cpp:445
#9  0x00007ffff787eb54 in cv::dnn::dnn4_v20201117::ONNXImporter::ONNXImporter (this=0x7fffffffd9f0, net=..., onnxFile=0x7fffffffdc30 "../RM.onnx") at /home/alexander/Projects/OpenCV/opencv-master/modules/dnn/src/onnx/onnx_importer.cpp:82
#10 0x00007ffff7898d70 in cv::dnn::dnn4_v20201117::readNetFromONNX (onnxFile="../RM.onnx") at /home/alexander/Projects/OpenCV/opencv-master/modules/dnn/src/onnx/onnx_importer.cpp:1891
```

**Comment by asmorkalov** [comment]
I confirm that #18845 fixes the reported issue in 3.4.
